index.ts

```diff
@@ -112,273 +113,162 @@ async function getTransactions(categoryLookup: {[key: string]: Category}, payeeL
+      await updateSheetValues(auth, process.env.SPREADSHEET_ID, process.env.ACCOUNTS_BALANCES_RANGE, accountBalances.map((a) => [a.name, a.balance]), "Account Balances");
     }
 
     if (!noCalcCategoryStats || !noEarMarkedTransactions) {
-      console.log("🏦 loading categories")
       const categories = await loadCategories();
-      console.log("🏦 loading transactions")
-      const ts = (await getTransactions(categories, await loadPayees())).map((t) => normalizeTransactionByCadence(t));;
+      const transactions = await getTransactions(categories, await loadPayees());
 
       if (!noCalcCategoryStats) {
-        console.log("🏦 Calculating category stats");
-        const categoryStats = calculateCategoryStats(categories, ts);
-        const categoryNames = Object.keys(categoryStats).sort();
-        const categoryStatsCsv = categoryNames.map((name: string) => categoryStats[name]).map((c: CategoryStats) => [c.name, c.group, c.average, c.weighted_average, c.budgeted]);
-        const googleAuth = await authorize();
-        await updateValues(googleAuth, process.env.SPREADSHEET_ID, process.env.SPREADSHEET_STATS_RANGE, categoryStatsCsv);
+        const categoryStats = calculateCategoryStats(categories, transactions);
+        await updateSheetValues(auth, process.env.SPREADSHEET_ID, process.env.SPREADSHEET_STATS_RANGE, Object.values(categoryStats).map((c) => [c.name, c.group, c.average, c.weighted_average, c.budgeted]), "Category Stats");
       }
 
       if (!noEarMarkedTransactions) {
-        console.log("🏦 Generating earmarked transactions");
-        const googleAuth = await authorize();
-        await updateValues(googleAuth, process.env.SPREADSHEET_ID, process.env.SPREADSHEET_EARMARKED_TRANSACTIONS_RANGE, generateEarmarkedTransactions(ts));
+        const earmarkedTransactions = generateEarmarkedTransactions(transactions);
+        await updateSheetValues(auth, process.env.SPREADSHEET_ID, process.env.SPREADSHEET_EARMARKED_TRANSACTIONS_RANGE, earmarkedTransactions, "Earmarked Transactions");
       }
     }
 
-    console.log("🎉 Done");
+    const currentDate = new Date();
+    const priorMonth = new Date(currentDate.getFullYear(), currentDate.getMonth() - 1, 1).toISOString().slice(0, 7);
+    const currentMonth = currentDate.toISOString().slice(0, 7);
+    const priorMonthData = await getMonthData(priorMonth);
+    const currentMonthData = await getMonthData(currentMonth);
+
+    await updateSheetValues(auth, process.env.SPREADSHEET_ID, process.env.PRIOR_MONTH_RANGE, priorMonthData, "Prior Month Data");
+    await updateSheetValues(auth, process.env.SPREADSHEET_ID, process.env.CURRENT_MONTH_RANGE, currentMonthData, "Current Month Data");
+
+    console.log("Data synchronization completed.");
     process.exit();
   },
 });
 
-// Run the command
-run(mainCommand, process.argv.slice(2)).catch(console.error);
\ No newline at end of file
+// Run the Command
```

Comment:
```suggestion
```
I'm just curious, why is `Command` upper case? I don't think the comment provides any value; I've committed it carelessly when I used CoPilot to help me setup `cmd-ts`.